RubberduckTests/Inspections/UnreachableCaseInspectionTests.cs

```diff
@@ -0,0 +1,2398 @@
+        {
+            var result =ValueFactory.Create(input, convertToTypeName);
+            Assert.IsNotNull(result, $"Type conversion to {convertToTypeName} return null interface");
+            Assert.AreEqual("Yahoo", result.ValueText);
+        }
+
+        [TestCase("NaN", "String")]
+        [Category("Inspections")]
+        public void UciUnit_ConvertToNanType(string input, string convertToTypeName)
+        {
+            var result = ValueFactory.Create(input, convertToTypeName);
+            Assert.IsNotNull(result, $"Type conversion to {convertToTypeName} return null interface");
+            Assert.AreEqual("NaN", result.ValueText);
+        }
+
+        [TestCase("10.51_*_11.2?Currency", "117.712", "Currency")]
+        [TestCase("10.51?Currency_*_11.2", "117.712", "Currency")]
+        [Category("Inspections")]
+        public void UciUnit_MathOpCurrency(string operands, string expected, string typeName)
+        {
+            var result = TestBinaryOp(MathSymbols.MULTIPLY, operands, expected, typeName);
+            Assert.AreEqual(typeName, result.TypeName);
+        }
+
+        [TestCase("10.51?Long_*_11.2", "123.2", "Double")]
+        [TestCase("10.51?Integer_*_11.2", "123.2", "Double")]
+        [TestCase("10.51?Byte_*_11.2", "123.2", "Double")]
+        [TestCase("10.51?Double_*_11.2", "117.712", "Double")]
+        [TestCase("10_*_11.2", "112", "Double")]
+        [TestCase("11.2_*_10", "112", "Long")]
+        [TestCase("10.51_*_11.2", "117.712", "Double")]
+        [TestCase("10.51?Single_*_11.2?Single", "117.712", "Single")]
+        [TestCase("10.51?Currency_*_11.2?Currency", "117.712", "Single")]
+        [TestCase("10_*_11", "110", "long")]
+        [TestCase("True_*_10", "-10", "Long")]
+        [TestCase("10_*_True", "-10", "Long")]
+        [Category("Inspections")]
+        public void UciUnit_Multiplication(string operands, string expected, string typeName)
+        {
+            TestBinaryOp(MathSymbols.MULTIPLY, operands, expected, typeName);
+        }
+
+        [TestCase("10_/_2", "5", "Long")]
+        [TestCase("2_/_10", "0", "Long")]
+        [TestCase("10_/_11.2", "0.89285", "Double")]
+        [TestCase("11.2_/_10", "1.12", "Double")]
+        [TestCase("10.51_/_11.2", "0.93839286", "Double")]
+        [TestCase("10_/_11", "1", "Long")]
+        [TestCase(@"""10.51""_/_11.2", "0.93839286", "Double")]
+        [TestCase("True_/_10.5", "-0.0952", "Double")]
+        [TestCase("10.5_/_True", "-10.5", "Double")]
+        [Category("Inspections")]
+        public void UciUnit_Division(string operands, string expected, string typeName)
+        {
+            TestBinaryOp(MathSymbols.DIVIDE, operands, expected, typeName);
+        }
+
+        [TestCase("10.51_+_11.2", "21.71", "Double")]
+        [TestCase("10_+_11.2", "21.2", "Double")]
+        [TestCase("11.2_+_10", "21.2", "Double")]
+        [TestCase("10_+_11", "21", "Long")]
```

Comment:
That's not right. It should be `Integer`, not `Long`. By default, implied integrals are dimensioned using `Integer` first, then `Long` if it's too big to fit the `Integer`.

Reply:
I think you are describing what VBA does....However, this inspection avoids all that type of manipulation and evaluates every integer-type as `Long` with constraints on the min and max values depending on the actual type of the Select Case Expression.  The inspection is simply looking for unreachable code - rather than trying to replicate VBA's 'math engine'.